src/main/scala/mesosphere/marathon/event/http/HttpEventActor.scala

```diff
@@ -2,53 +2,91 @@ package mesosphere.marathon.event.http
   implicit val timeout = HttpEventModule.timeout
+  val pipeline: HttpRequest => Future[HttpResponse] = addHeader("Accept", "application/json") ~> sendReceive
+  var limiter = Map.empty[String, EventNotificationLimit].withDefaultValue(NoLimit)
 
-  val pipeline: HttpRequest => Future[HttpResponse] = (
-    addHeader("Accept", "application/json")
-    ~> sendReceive)
+  //metrics
+  val metricOutstandingRequests = metrics.counter("service.mesosphere.marathon.event.http.outstanding-requests")
+  val metricSendEvent = metrics.meter("service.mesosphere.marathon.event.http.send")
+  val metricDroppedEvent = metrics.meter("service.mesosphere.marathon.event.http.dropped-events")
+  val metricResponseTime = metrics.timer("service.mesosphere.marathon.event.http.response-time")
 
   def receive: Receive = {
-    case event: MarathonEvent =>
-      broadcast(event)
-    case _ =>
-      log.warning("Message not understood!")
+    case event: MarathonEvent     => broadcast(event)
+    case NotificationSuccess(url) => limiter += url -> NoLimit
+    case NotificationSlow(url)    => limiter += url -> limiter(url).nextFailed
+    case NotificationFailed(url)  => limiter += url -> limiter(url).nextFailed
+    case _                        => log.warning("Message not understood!")
   }
 
   def broadcast(event: MarathonEvent): Unit = {
+    metricSendEvent.mark()
     log.info("POSTing to all endpoints.")
-    (subscribersKeeper ? GetSubscribers).mapTo[EventSubscribers].foreach {
-      _.urls.foreach { post(_, event) }
+    val me = self
+    (subscribersKeeper ? GetSubscribers).mapTo[EventSubscribers].foreach { subscribers =>
+      val (active, limited) = subscribers.urls.partition(limiter(_).notLimited)
+      if (limited.nonEmpty) log.info(s"Will not send event ${event.eventType} to unresponsive hosts: $limited")
+      metricDroppedEvent.mark(limited.size)
+      active.foreach(post(_, event, me))
     }
   }
 
-  def post(urlString: String, event: MarathonEvent): Unit = {
-    log.info("Sending POST to:" + urlString)
+  def post(url: String, event: MarathonEvent, eventActor: ActorRef): Unit = {
+    log.info("Sending POST to:" + url)
 
-    val request = Post(urlString, eventToJson(event))
+    metricOutstandingRequests.inc()
+    val request = Post(url, eventToJson(event))
     val response = pipeline(request)
+    val start = currentTimeMillis()
 
     response.onComplete {
+      case _ =>
+        metricOutstandingRequests.dec()
+        metricResponseTime.updateMillis(currentTimeMillis() - start)
+    }
+    response.onComplete {
       case Success(res) =>
-        if (res.status.isFailure)
-          log.warning(s"Failed to post $event to $urlString")
-
-      case Failure(t) =>
-        log.warning(s"Failed to post $event to $urlString", t)
+        if (res.status.isFailure) log.debug(s"No success response for post $event to $url")
+        val inTime = (currentTimeMillis() - start) < conf.slowConsumerTimeout
```

Comment:
So HTTP-level failures (500, ...) are not treated as failures? I think it is more polite doing so because if the receiving end is overloaded, we will desist. Of course, this changes the behavior a bit.

Then, (because we back off and prevent spamming the logs), I would at least log the HTTP status code + status string as warning.